src/main/java/org/netcracker/educationcenter/elasticsearch/database/operations/ElasticsearchOperations.java

```diff
@@ -0,0 +1,118 @@
+package org.netcracker.educationcenter.elasticsearch.database.operations;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.action.update.UpdateResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.netcracker.educationcenter.elasticsearch.Connection;
+
+import java.io.IOException;
+
+/**
+ * Abstract class for Elasticsearch Database operations (Insert, Get, Delete, Update)
+ *
+ * @author Mikhail Savin
+ */
+public abstract class ElasticsearchOperations {
+    private static final Logger LOG = LogManager.getLogger();
+
+    /**
+     * Current connection instance
+     */
+    private Connection connection;
+
+    /**
+     * Constructor with given connection to interact with ES DB.
+     *
+     * @param connection current connection
+     */
+    public ElasticsearchOperations(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * Inserts given JSON String (with model) into the ES Database
+     *
+     * @param jsonString JSON String of the inserted object
+     * @param id id of the inserted object
+     * @param index index of the inserted model
+     */
+    public void insert(String jsonString, String id, String index) {
+        IndexRequest indexRequest = new IndexRequest(index)
+                .id(id).source(jsonString, XContentType.JSON);
+        try {
+            IndexResponse indexResponse = connection.getRestHighLevelClient()
+                    .index(indexRequest, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            LOG.error(e);
+        }
+    }
+
+    /**
+     * Gets model's JSON as a String by its id and index.
+     *
+     * @param id searched JSON id
+     * @param index searched JSON index
+     * @return searched JSON as a String
+     * @throws IOException if something wrong with get method
+     */
+    public String getById(String id, String index) throws IOException {
```

Comment:
Why this method throws IOException but the others don't?

Reply:
Because of "throw e;" line 76.
```suggestion
    public String getById(String id, String index) throws IOException {
        GetRequest getJiraIssuesRequest = new GetRequest(index, id);
        GetResponse getResponse = null;
        try {
            getResponse = connection.getRestHighLevelClient()
                    .get(getJiraIssuesRequest, RequestOptions.DEFAULT);
        } catch (ElasticsearchException | IOException e) {
            LOG.error(e);
        }
        if (getResponse != null && getResponse.isExists()) {
            return getResponse.getSourceAsString();
        } else {
            return "Document was not found";
        }
    }
```
I have removed it.
Is only logging OK for exception handling in this situation?